compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -1805,8 +1805,27 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
                         || self.tcx.is_builtin_derive(def_id)
                 })
                 .filter_map(|def_id| self.tcx.impl_trait_ref(def_id))
-                // Avoid mentioning type parameters.
-                .filter(|trait_ref| !matches!(trait_ref.self_ty().kind(), ty::Param(_)))
+                .filter(|trait_ref| {
+                    // Avoid mentioning type parameters.
```

Comment:
Was it intentional to have this comment twice?

Reply:
No, lol